components/service/glean/src/main/java/mozilla/components/service/glean/net/HttpPingUploader.kt

```diff
@@ -21,6 +23,29 @@ class HttpPingUploader(configuration: Configuration) : PingUploader {
     private val config = configuration
     private val logger = Logger("glean/HttpPingUploader")
 
+    /**
+     * Log the contents of a ping to the console, if configured to do so in
+     * config.logPings.
+     *
+     * @param path the URL path to append to the server address
+     * @param data the serialized text data to send
+     */
+    private fun logPing(path: String, data: String) {
+        if (config.logPings) {
+            // Parse and reserialize the JSON so it has indentation and is human-readable.
+            var indented = try {
+                var json = JSONObject(data)
+                json.toString(2)
+            } catch (e: JSONException) {
+                logger.debug("Exception parsing ping as JSON: $e") // $COVERAGE-IGNORE$
+                null // $COVERAGE-IGNORE$
+            }
+            indented?.let {
+                logger.debug("Glean ping to URL: ${path}\n$it")
+            }
```

Comment:
nit: The code may be a bit shorter if you just log from inside the `try` block without returning anything? :)